From 97de8780de730322435f56210695cbb52828c559 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Thu, 12 Mar 2020 19:26:03 +0200
Subject: [PATCH] [SW-10413] goya2: mask spi flash access offset

To protect from invalid access to offset outside of flash
we're masking the offset so only valid ranges will be accessed

Change-Id: I15fd9db001299c1320e8b45a5dea9a7ee0a93b32
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/spi_flash.c b/common/spi_flash.c
index 40ea726..e6a7dd9 100644
--- a/common/spi_flash.c
+++ b/common/spi_flash.c
@@ -30,6 +30,8 @@
 
 uint32_t spi_flash_readl(uint32_t offset)
 {
+	/* mask offset to valid range of flash */
+	offset &= FLASH_VALID_MASK;
 	uint32_t reg = base_readl(HL_FLASH_BASE + offset);
 
 	return reg;
@@ -37,6 +39,8 @@
 
 void spi_flash_writel(uint32_t val, uint32_t offset)
 {
+	/* mask offset to valid range of flash */
+	offset &= FLASH_VALID_MASK;
 	base_writel(val, HL_FLASH_BASE + offset);
 }
 
diff --git a/common/spi_flash.h b/common/spi_flash.h
index 9f10a09..52491a5 100644
--- a/common/spi_flash.h
+++ b/common/spi_flash.h
@@ -25,6 +25,7 @@
 
 #define SPI_FLASH_BAUDR_DIVISOR		2
 
+#define FLASH_VALID_MASK		0xFFFFFF
 #define FLASH_64KB_PADD			65536
 #define FLASH_FW_VERSION_SIZE		16
 #define FLASH_KEY_HASH_SIZE		160
